src/test/java/com/google/javascript/clutz/partial/missing_base.js

```diff
@@ -1,4 +1,4 @@
-goog.module('mising.extend');
+goog.module('missing.extend');
```

Comment:
this seems to be lacking a few cases:
- the recursive call in getImportRenameMap
- a file not containing a goog.module call, but top level assignments (maybe that's a Closure level error?)
- destructuring call `const {x} = goog.require(...);`
- top level `goog.require` where the symbol required is from a `goog.provide` style file, not from a `goog.module`

Reply:
> 4. We already have a test for referring to a type by its full name (`direct.ref.A`).

This is what I meant:

```js
// file A
goog.provide('ns.A');
/** @constructor */
ns.A = function() {...};

// file B
goog.module('mod');
const A = goog.require('ns.A');
```

Now if you compile file B in isolation, I think this code emits a bunch of `module$something` strings that might not match up with file A produces when compiled in isolation (as there's no module stuff involved there).

Reply:
Got it.  Talked it over with Rado, and there's currently no way to distinguish imports of goog.provide() files from goog.module().  We think it's outside the scope of this PR to fix that, so I've filed an issue https://github.com/angular/clutz/issues/596.